core-bundles/meta-model/org.eclipse.vorto.core/src/org/eclipse/vorto/core/api/model/ModelConversionUtils.java

```diff
@@ -104,6 +101,18 @@ public static FunctionblockModel convertToFlatHierarchy(FunctionblockModel fbm)
     return fbm;
   }
 
+  private static List<Property> removeDuplicates(List<Property> allProperties) {
+    return allProperties.stream()
+            .filter(filterByTypeAndName(p -> p.getName() + "_" + p.getType().getClass().getName()))
+            .collect(Collectors.toList());
+  }
+
+  public static <T> Predicate<T> filterByTypeAndName(
+          Function<? super T, ?> generateKey) {
+    Map<Object, Boolean> insertedPropertiesMap = new ConcurrentHashMap<>();
```

Comment:
Why did you use ConcurrentHashMap here?

Reply:
Because non concurrent map would not work with parallel streams. But you are right- where its used now the stream is sequential. Do you want me to replace it with a non concurrent map?

Reply:
I think a non-concurrent HashMap would be more efficient in terms of performance.